File: module/csharp/Module.cs
using ErpCore.Component.Host;
namespace MyModule;

public class ModuleImpl : IModule {
    public static string Start() {
        Host.Log("info", "module started");
        return """{"success":true,"data":null,"message":"ok"}""";
    }
    public static string Stop() => """{"success":true}""";
    public static string CallRoutine(string name, string input) =>
        """{"success":false,"message":"not implemented"}""";
    public static string CallEndpoint(string name, string method, string path, string body, string headers) =>
        """{"success":false,"message":"not implemented"}""";
    public static string CallEvent(string name, string trigger, string entity, string data) =>
        """{"success":false,"message":"not implemented"}""";
}

File: module/typescript/module.ts
// @ts-ignore
import { log } from "erp-core:component/host";

export const start = (): string => {
    log("info", "module started");
    return JSON.stringify({ success: true, data: null, message: "ok" });
};
export const stop = (): string => JSON.stringify({ success: true });
export const callRoutine = (_name: string, _input: string): string =>
    JSON.stringify({ success: false, message: "not implemented" });
export const callEndpoint = (_name: string, _method: string, _path: string, _body: string, _headers: string): string =>
    JSON.stringify({ success: false, message: "not implemented" });
export const callEvent = (_name: string, _trigger: string, _entity: string, _data: string): string =>
    JSON.stringify({ success: false, message: "not implemented" });
